src/System.Runtime/src/System/HashCode.cs

```diff
@@ -0,0 +1,406 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    public struct HashCode
```

Comment:
It might be good to comment here that we're using xxHash32 and link back to it for future reference.